source/ucumLhcUtils.js

```diff
@@ -165,43 +165,48 @@ export class UcumLhcUtils {
   *
   * @param {Object} fromUnit - The unit to convert from.
   * @param {Object} toUnit - The unit to convert to.
-  * @returns {ConversionType} conversionType - The type of conversion as a string. Possible values are:
-  * 'normal', 'mol->mass', 'mass->mol', 'eq->mass', 'mass->eq', 'eq->mol', 'mol->eq', 'mol<->mol', 'eq<->eq'.
+  * @returns {ConversionType} conversionType - The type of conversion as a string.
   */
   detectConversionType(fromUnit, toUnit) {
     /** @type {ConversionType} */
     let conversionType = 'normal';
 
-    // if neither unit is not a molar unit, return 'normal'
+    /** Note: 
+     * isMolarUnit() will return true for both molar and equivalent units 
+     * because equivalent units are a subset of molar units. When checking 
+     * for eq<->mol conversions, we need explicitly check that one unit is
+     * an equivalent unit and the other is a molar unit.
+     * 
+     * For this reason, the order of the below checks matters and we need to
+     * check for eq<->mol conversions before mol<->mol conversions.
+     */ 
+
+    // if neither unit is a molar/eq unit, return 'normal'
     if (!(fromUnit.isMolarUnit() || toUnit.isMolarUnit())) {
       return conversionType;
     }
      
-    // Otherwise, handle mol <-> eq <-> mass conversions
-    if (fromUnit.isMolarUnit() && toUnit.isMolarUnit() && !fromUnit.isEquivalentUnit() && !toUnit.isEquivalentUnit()) {
-      conversionType = 'mol<->mol';
-    } else if (fromUnit.isEquivalentUnit() && toUnit.isEquivalentUnit()) {
+    // handle eq <-> eq conversions
+    if (fromUnit.isEquivalentUnit() && toUnit.isEquivalentUnit()) {
       conversionType = 'eq<->eq';
     }
     // handle eq <-> mol/mass conversions
-    else if (fromUnit.isEquivalentUnit() || toUnit.isEquivalentUnit()) {
-      if (fromUnit.isEquivalentUnit() && toUnit.isMolarUnit()) {
-        conversionType = 'eq->mol';
-      } else if (fromUnit.isMolarUnit() && toUnit.isEquivalentUnit()) {
-        conversionType = 'mol->eq';
-      } else if (fromUnit.isEquivalentUnit() && !toUnit.isMolarUnit()) {
-        conversionType = 'eq->mass';
-      } else if (!fromUnit.isMolarUnit() && toUnit.isEquivalentUnit()) {
-        conversionType = 'mass->eq';
-      }
-    } 
+    else if (fromUnit.isEquivalentUnit() && toUnit.isMolarUnit()) {
+      conversionType = 'eq->mol';
+    } else if (fromUnit.isMolarUnit() && toUnit.isEquivalentUnit()) {
+      conversionType = 'mol->eq';
+    } else if (fromUnit.isEquivalentUnit()) {
+      conversionType = 'eq->mass';
+    } else if (toUnit.isEquivalentUnit()) {
+      conversionType = 'mass->eq';
+    }
     // handle mol <-> mass conversions
-    else {
-      if (fromUnit.isMolarUnit() && !toUnit.isMolarUnit()) {
-        conversionType = 'mol->mass';
-      } else if (!fromUnit.isMolarUnit() && toUnit.isMolarUnit()) {
-        conversionType = 'mass->mol';
-      }
+    else if (fromUnit.isMolarUnit() && toUnit.isMolarUnit()) {
+      conversionType = 'mol<->mol';
```

Comment:
Wouldn't mol<->mol and eq<->eq just be "normal"?

Reply:
Hmm, yes that would make sense. In hindsight, mol -> mmol or eq ->meq are valid conversions and should be handled normally. I'll convert those cases to normal cases